packages/material-tree-renderer/test/services/property.util.test.ts

```diff
@@ -352,6 +352,77 @@ describe('Property util', () => {
     ]);
   });
 
+    test('expect to retrieve properties from object with array object by using keyword anyOf twice', () => {
+        const schema: JsonSchema = {
+            definitions: {
+                a: { anyOf: [{$ref: '#/definitions/b'}] },
+                b: { type: 'object', properties: { foo: { type: 'string' } } }
+            },
+            type: 'object',
+            properties: {
+                elements: {
+                    type: 'array',
+                    items: {
+                        anyOf: [
+                            { $ref: '#/definitions/a' },
+                        ]
+                    }
+                }
+            }
+        };
+
+        const properties = findContainerProperties(schema, schema, false);
+        expect(properties).toMatchObject([
+            {
+                property: 'elements',
+                label: 'b',
+                schema: {
+                    properties: {
+                        foo: { type: 'string' }
+                    }
+                }
+            }
+        ]);
+    });
+
+    test('expect to retrieve properties from object with array object by using keyword anyOf recursive', () => {
+        const schema: JsonSchema = {
+            definitions: {
+                a: {
+                    anyOf: [
+                        {$ref: '#/definitions/b'},
+                        {$ref: '#/definitions/a'}
+                    ]
+                },
+                b: { type: 'object', properties: { foo: { type: 'string' } } }
```

Comment:
Maybe we can make this example a bit more complex by adding the `anyOf` definition of `a` into `b` 's properties. We can create circular references between definitions, then we can check whether any of these definitions are visited more than once. What do you think?